src/util/ajax.js

```diff
@@ -130,8 +130,21 @@ function sameOrigin(url) {
 
 const transparentPngUrl = 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAAC0lEQVQYV2NgAAIAAAUAAarVyFEAAAAASUVORK5CYII=';
 
-export const getImage = function(requestParameters: RequestParameters, callback: Callback<HTMLImageElement>): Cancelable {
-    // request the image with XHR to work around caching issues
+export const getImage = function(requestParameters: RequestParameters, callback: Callback<HTMLImageElement>, avoidXHR?: boolean): Cancelable {
+    // if we know for sure that the tile is cached for a long time, avoid XHR for better performance
+    // https://github.com/mapbox/mapbox-gl-js/issues/6643
+    if (avoidXHR && requestParameters.headers === undefined && requestParameters.credentials === undefined) {
+        const img: HTMLImageElement = new window.Image();
+        const url = requestParameters.url;
+        if (!sameOrigin(url)) {
+            img.crossOrigin = 'Anonymous';
+        }
+        img.onload = () => callback(null, img);
+        img.src = url;
+        return {cancel: () => { img.src = transparentPngUrl; }};
```

Comment:
Shouldn't `callback(err)` be handled here as well?

Reply:
I think an [onerror](https://developer.mozilla.org/en-US/docs/Web/API/HTMLImageElement#Errors) handler would be enough.

Reply:
Good point, added.